lib/config/validation.ts

```diff
@@ -135,6 +146,17 @@ export async function validateConfig(
         message: `The "${key}" object can only be configured at the top level of a config but was found inside "${parentPath}"`,
       });
     }
+    if (!isGlobalConfig) {
+      if (optionGlobals.has(key)) {
+        // token is a global config option and a field of repo config option hostRules.encrypted
+        if (!(key === 'token' && parentPath?.includes('hostRules'))) {
```

Comment:
What about username and password - are they different?

Reply:
Missed them, didn't know they could be configured without a parent.

Reply:
I looked for all other possible options as well. Here's the result:

By "other possible options" I mean the repo config options which can have fields whose  whose children can have fields same as another `globalOnly` option.

Except for `secrets`, `encrypted`, `hostRules` no other config option will have such fields. Other options either are either validated separately or they have strict restrictions as to which fields are allowed in them.

Among these three options:
- `hostRules` is already handled in this PR.

- `secrets` is validate separately ie. its children/fields won't be recursively passed inside validateConfig` for re-validation. So we don't need extra check for its options.

- Fields inside `encrypted` needs to be handled based on its parent.
If we handle the rest of the options well, `encrypted` doesn't needs to be handled separately.